packages/plugin-core/src/commands/GotoNote.ts

```diff
@@ -93,6 +94,30 @@ export class GotoNoteCommand extends BasicCommand<CommandOpts, CommandOutput> {
               };
             }
           }
+          // handle frontmatter tags
+          // First, check if the current line has a single line tag like `tags: foo`
+          const singleTag = currentLine.match(NoteUtils.RE_FM_TAGS)?.groups?.singleTag;
+          if (singleTag) {
+            return {
+              alias: singleTag,
+              value: `${TAGS_HIERARCHY}${singleTag}`,
+            };
+          }
+          // Otherwise, check the text up to the selection for frontmatter tags
+          // list. The last tag in that is either the selection, or something
+          // before the selection. If it is the selection, then the user has the
+          // tag selected. This is not perfect since the user could be selecting
+          // something past the frontmatter that just happens to match the last
+          // tag, but that's an edge case that's not disruptive.
+          const frontmatterText = editor?.document.getText(new Range(0, 0, selection.start.line, currentLine.length));
+          const selectedItem = currentLine.match(NoteUtils.RE_FM_LIST_ITEM)?.groups?.listItem;
```

Comment:
as mentioned above, this regex will mix inline list case. wondering if it would be less error prone to just grab the text and run [string2note](https://github.com/dendronhq/dendron/blob/feat/frontmatter-tags/packages/common-server/src/filesv2.ts#L138:L138) to parse for tags? as an optimization, we could just parse the frontmatter instead of the entire note

Reply:
we could refactor [this](https://github.com/dendronhq/dendron/blob/feat/frontmatter-tags/packages/plugin-core/src/services/NoteSyncService.ts#L21:L21) into a general utility

Reply:
The problem is, a parsed note has no information about what line of text each frontmatter property came from. The `matter` library we use doesn't provide that information, just the parsed YAML. I considered adding a unist YAML parser to parse the contents of the frontmatter into a tree, but opted to just parse with the regex for now since it's simpler.

But I'm thinking now that it might actually be a better choice to use the actual parser, since the YAML is more complex that I initially thought and my implementation doesn't do a great job of parsing it.